Manga chan/list parser

MariaAfanasyeva opened the pull request:
### Описание

Сделала list-parser для manga-chan

-------

<!-- Нужно указать #номер_issue для того, что бы issue автоматически заркывались после мерджа -->
closes #105

MariaAfanasyeva committed:
Created Anibel list parser with scrapy

MariaAfanasyeva committed:
Merge remote-tracking branch 'origin/develop' into develop

MariaAfanasyeva committed:
Merge remote-tracking branch 'origin/develop' into develop

MariaAfanasyeva committed:
Merge remote-tracking branch 'origin/develop' into develop

MariaAfanasyeva committed:
Merge remote-tracking branch 'origin/develop' into develop

MariaAfanasyeva committed:
Merge remote-tracking branch 'origin/develop' into develop

MariaAfanasyeva committed:
Added manga_chan parser to parsers.py

dhvcc commented:
Лучше бы убрать парсер Anibel из этого ПРа
Плюс лучше бы ребейснуть мерджи девелопа, что бы по итогу вышел 1 коммит
@MariaAfanasyeva

NikDark commented on apps/parse/manga_chan/list_parser/manga_spider.py:
```diff
@@ -0,0 +1,87 @@
+            source_url = MANGA_CHAN_URL + response.xpath(SOURCE_URL_TAG).extract_first("")
+            genres = response.xpath(GENRES_TAG).extract()
+            image = response.xpath(IMAGE_TAG).extract_first("")
+            titles = full_title.split(" (")
+            alt_title = titles[0]
+            title = titles[-1]
```
Давай может получать `alt_title` через XPath `//div[@class = "manga_row1"]//a[@class = "title_link"]/@title`
Также full_title состоит из `alt_title` + 'пробел' + '(' + `title` + ')'
Не нужна будет проверка ниже `if alt_title != title`

NikDark commented on apps/parse/manga_chan/list_parser/pipelines.py:
```diff
@@ -0,0 +1,42 @@
+        genres = data.pop("genres")
+        source_url = data.pop("source_url")
+
+        if not title:
+            message = f"Error processing {data}: No title name was set"
+            spider.logger.error(message)
+            raise KeyError(message)
```
Я думаю что после изменений выше, этот код будет не нужен, потому что у нас как минимум название будет содержать `alt_title`. А если нету даже `alt_title` то это скорее всего и на сайте нету. Ну, можно на крайний случай добавить проверку в `manga_spider`

NikDark commented on apps/parse/parsers.py:
```diff
@@ -30,6 +32,19 @@ def readmanga_parser(settings=None, logger=None):
+            **get_project_settings(),
+            **(settings if settings else {}),
+        }
+    )
```
Объедини пожалуйста `readmanga_parser` и `manga_chan_parser`, а то здесь только одна переменная отличается
Можно написать dict который будет содержать в себе 
```python
spiders = {
    Manga.SOURCE_MAP.get("https://manga-chan.me") : {
        "spider": MangaChanSpider,
        "settings": MANGA_CHAN_SETTINGS_PATH
    }
}
```

dhvcc replied:
Давай это позже обсудим, в контексте абстракции не только этой функции

NikDark commented on apps/parse/parsers.py:
```diff
@@ -53,4 +68,7 @@ def mangalib_parser(settings=None, logger=None):
         IMAGE_PARSER: mangalib_images_info,
     },
+    Manga.SOURCE_MAP.get("https://manga-chan.me"): {
+        LIST_PARSER: manga_chan_parser,
```
ну и здесь доставать из дикта нашего паука
- также не забудь поменять в `ReadManga`
```py
spiders.get(Manga.SOURCE_MAP.get("https://manga-chan.me"))
```

NikDark reviewed (commented):
По поводу последнего объединения это всего лишь предложение
Можешь использовать и своё какое-то решение
Можно написать замыкания еще наверное
В общем что захочешь

MariaAfanasyeva committed:
Added ALT_TITLE_TAG

Updated readmanga images_parser according to new HTML

Removed somewhat unsafe ast.literal_eval and replaced it with json.loads

Added ec2-deploy workflow

Deploy updates

Created popularity field for Readmanga

MariaAfanasyeva committed:
Added popularity field

dhvcc commented on .github/workflows/ec2-deploy.yml:
```diff
@@ -0,0 +1,31 @@
+name: EC2 deploy
+
+on:
+  workflow_run:
```
А это чего тут делает?)

dhvcc commented on .gitignore:
```diff
@@ -160,3 +160,4 @@ staticfiles/
 redis-data
 redis.conf
 .vim
+.nohup.out
```
Ты, видимо, подтянула мой код, но не коммиты

dhvcc commented on apps/parse/models.py:
```diff
@@ -83,10 +84,17 @@ class Manga(BaseModel):
     updated_detail = models.DateTimeField(blank=True, null=True)
     updated_chapters = models.DateTimeField(blank=True, null=True)
+    # Popularity = index of the manga in the catalogue
+    popularity = models.IntegerField(default=0)
```
А это код Никиты

dhvcc commented on manga_reader/settings.py:
```diff
@@ -20,6 +20,7 @@
 SECRET_KEY = os.getenv("SECRET_KEY")
 DEBUG = int(os.getenv("DEBUG", 1))
 ALLOWED_HOSTS = os.getenv("ALLOWED_HOSTS", "*").split(" ")
+SECURE_PROXY_SSL_HEADER = ("HTTP_X_FORWARDED_PROTO", "https")
```
мой код, но не коммит

dhvcc commented on pyproject.toml:
```diff
@@ -30,6 +30,8 @@ django-admin-easy = "0.6.1"
 pyppeteer = "^0.2.5"
 sentry-sdk = "^1.3.0"
+gunicorn = "^20.1.0"
+Werkzeug = "0.16.0"
```
:eyes:

dhvcc reviewed (changes requested):
Надо починить историю коммитов + пара микро-фиксов
